### ISSUE TITLE
Time in Service not Initialized for all Valid Personnel

### ISSUE THREAD

**Issue by Wargamer777** [MegaMek/mekhq#1847]
.47.7 Release
Windows 10
Java 8 Version 251

When adding new personnel to mekhq campaign, you can no longer click on them and see their details on the right side panel.  You can still right click on them and interact, but it isn't apparent in the view.

Examples in attached campaign file would be Leutnant Delores Valentine or Candice Sunny. 

[The Blood Hound Marauders30410601.zip](https://github.com/MegaMek/mekhq/files/4893142/The.Blood.Hound.Marauders30410601.zip)

**Comment by Windchild292** [comment]
For the meantime, disabling Time in Rank will fix the issues. However, it does appear that there are missing log entries and recruitment dates for certain personnel that I will look into.

**Comment by Windchild292** [comment]
This campaign file did help me find the Time in Service cause of the Time in Rank issue. I've renamed it to fit the bug that needs to be fixed since the Time in Rank issue has been fixed by #1834

**Comment by Wargamer777** [comment]
Cool, great that it helped.  Just to note, I was going through the Campaign Options since I hadn't played in a while and had upgraded from a previous build.  Basically, I caused the problem because I was "turning on" new options that have been made available since the last version of MekHQ I had played.  The Time In Rank was one specifically that I DID turn on.

**Comment by Windchild292** [comment]
The issue was that I made a mistake when coding Time in Rank, and didn't catch the issue while testing or playing myself (it was added in 0.47.6 and I use it in my games, so I'm surprised it went uncaught until this week).

It is a software issue, so thanks for reporting it.

**Comment by JonhyBurner** [comment]
Just to add that I also have been having issues with Time in Service, but I never used Time in Rank.

It might be related to the update, since my oldest save is a few weeks old, but it already had the issue, I just hadn't caught up with it. I did added it to an already going campaign.

I can interact just fine with all the personnel, but some don't have the log entry for joining my unit and no Time in Service. The save file does have recruitment date, and if I uncheck and recheck the Time in Service, the time appears and is counted correctly. That is for personnel already in the save. 

New personnel behaves weirdly. They can be hired (No recruit log entry, no Time in Service). If I try the check/uncheck trick, the Time in Service appears, but it counts from exactly one year before the current campaign date. If I advance a few days with a recent hire with no Time in Service (I tried a month) and do the trick, it again counts from a year before the current date. But if I give an award and do the trick, the time gets counted correctly (this might work with any log entry).

I also noticed that prisoners logs are different. They get "Demoted to prisoner" before being added to the prisoners roster. I don't know if this is intended behavior.
